testing/s3mock/src/main/java/org/projectnessie/s3mock/AwsChunkedInputStream.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.s3mock;
+
+import com.google.common.base.Preconditions;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+
+final class AwsChunkedInputStream extends InputStream {
+  private final InputStream input;
+  private AwsChunkedState state = AwsChunkedState.EXPECT_HEADER;
+  private int chunkLen;
+
+  // https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html#sigv4-chunked-body-definition
+
+  // Interestingly, for a 1MB upload, the AWS sync client uses 'aws-chunked' content-encoding, but
+  // the AWS async client does not.
+
+  AwsChunkedInputStream(InputStream input) {
+    this.input = input;
+  }
+
+  enum AwsChunkedState {
+    EXPECT_HEADER,
```

Comment:
"Header" in this context is a bit misleading imo, maybe rename to `METADATA` to align with `DATA`.